darshan-util/pydarshan/darshan/cli/summary.py

```diff
@@ -0,0 +1,358 @@
+    sns_plot.set_xticklabels(sns_plot.get_xticklabels(), rotation=0, size=6.5)
+    sns_plot.set_yticklabels(sns_plot.get_yticks(), rotation=0, size=6.5)
+    sns_plot.set_xlabel('Rank', size=8)
+    sns_plot.set_ylabel('Time (seconds)', size=8)
+    #sns.despine();
+    pyplot.savefig(buf, format='png', bbox_inches='tight')
+    buf.seek(0)
+    encoded.append(base64.b64encode(buf.read()))
+    return encoded
+  
+def main(args=None):
+
+    if args is None:
+        parser = argparse.ArgumentParser(description='')
+        setup_parser(parser)
+        args = parser.parse_args()
+
+    if args.debug:
+        print(args)
+
+    variables = {}
+    report = darshan.DarshanReport(args.input, read_all=True)
+    report.info()
+
+    #
+    # Setup template header variabels
+    #
+    variables['exe'] = report.metadata['exe']
+    variables['date'] = datetime.datetime.fromtimestamp(report.metadata['job']['start_time'], pytz.utc)
+    variables['jid'] = report.metadata['job']['jobid']
+    variables['uid'] = report.metadata['job']['uid']
+    variables['nprocs'] = report.metadata['job']['nprocs']
+    etime = int(report.metadata['job']['end_time'])
+    stime = int(report.metadata['job']['start_time'])
+    if etime > stime:
+        variables['runtime'] = etime - stime + 1
+    else:
+        variables['runtime'] = 0
+
+    if 'POSIX' in report.modules:
+        posix_df = report.records['POSIX'].to_df()
+    else:
+        posix_df = None
+
+    if 'MPI-IO' in report.modules:
+        mpiio_df = report.records['MPI-IO'].to_df()
+    else:
+        mpiio_df = None
+    
+    if 'STDIO' in report.modules:
+        stdio_df = report.records['STDIO'].to_df()
+    else:
+        stdio_df = None
+    
+    if 'APXC' in report.modules:
+        apxc_dict = report.records['APXC'].to_dict()
+    else:
+        apxc_dict = None
+
+    if 'APMPI' in report.modules:
+        apmpi_dict = report.records['APMPI'].to_dict()
+    else:
+        apmpi_dict = None
+
+    #
+    # Plot I/O cost
+    #
+    variables['plot_io_cost'] = plot_io_cost(posix_df, mpiio_df, stdio_df, int(variables['runtime']), int(variables['nprocs'])).decode('utf-8')
+
+    #
+    # Plot I/O counts
+    #
+    variables['plot_op_count'] = plot_op_count(posix_df, mpiio_df, stdio_df).decode('utf-8')
+
+    variables['fs_data'] = data_transfer_filesystem(report, posix_df, stdio_df)
+    apmpi_encoded = apmpi_process(apmpi_dict)
+    variables['apmpi_call_time'] = apmpi_encoded[0].decode('utf-8')
+    variables['apmpi_rank_totaltime'] = apmpi_encoded[1].decode('utf-8')
+
+    template_path = pkg_resources.path(darshan.templates, '')
+    with template_path as path:
+       loader = genshi.template.TemplateLoader(str(path))
+       template = loader.load('summary.html')
+
+       stream = template.generate(title='Darshan Job Summary', var=variables)
+       with open(args.output, 'w') as f:
+           f.write(stream.render('html'))
+           f.close()
```

Comment:
the file handle `f` is automatically closed by the context manager, so this explicit `close()` can be removed